fontir/src/glyph.rs

```diff
@@ -634,23 +634,96 @@ fn ensure_notdef_exists_and_is_gid_0(
         None => {
             trace!("Generate {} and make it gid 0", GlyphName::NOTDEF);
             glyph_order.set_glyph_id(&GlyphName::NOTDEF, 0);
-            let static_metadata = context.static_metadata.get();
-            let metrics = context
-                .global_metrics
-                .get()
-                .at(static_metadata.default_location());
-            let builder = GlyphBuilder::new_notdef(
-                static_metadata.default_location().clone(),
-                static_metadata.units_per_em,
-                metrics.ascender.0,
-                metrics.descender.0,
-            );
-            context.glyphs.set(builder.build()?);
+            let notdef = synthesize_notdef(context)?;
+            context.glyphs.set(notdef);
         }
     }
     Ok(())
 }
 
+/// Create a (possibly variable) .notdef glyph
+///
+/// * see <https://github.com/googlefonts/ufo2ft/blob/b3895a96ca/Lib/ufo2ft/outlineCompiler.py#L1666-L1694>
+/// * and <https://github.com/googlefonts/fontc/issues/1262>
+fn synthesize_notdef(context: &Context) -> Result<Glyph, BadGlyph> {
+    let static_metadata = context.static_metadata.get();
+    let global_metrics = context.global_metrics.get();
+
+    let mut builder = GlyphBuilder::new(GlyphName::NOTDEF);
+    let upem = static_metadata.units_per_em;
+    let width = (upem as f64 * 0.5).ot_round();
+    let default = global_metrics.at(static_metadata.default_location());
+    let default_outline = make_notdef_outline(upem, default.ascender.0, default.descender.0);
+    // TODO: Should this be None to match other glyphs' heights? This would deviate from ufo2ft
+    // See https://github.com/googlefonts/ufo2ft/blob/b3895a96/Lib/ufo2ft/outlineCompiler.py#L1656-L1
+    let height = Some(default.ascender.0 - default.descender.0);
+
+    builder.try_add_source(
+        static_metadata.default_location(),
+        GlyphInstance {
+            width,
+            height,
+            contours: vec![default_outline],
+            ..Default::default()
+        },
+    )?;
+    if static_metadata.axes.is_empty() {
+        return builder.build();
+    }
+
+    for location in static_metadata.variation_model.locations() {
+        let ascender = global_metrics.get(GlobalMetric::Ascender, location);
+        let descender = global_metrics.get(GlobalMetric::Descender, location);
+        if (ascender, descender) == (default.ascender, default.descender) {
+            continue;
+        }
+        let outline = make_notdef_outline(upem, ascender.0, descender.0);
+
+        let instance = GlyphInstance {
+            width,
+            height: Some(ascender.0 - descender.0),
+            contours: vec![outline],
+            ..Default::default()
+        };
+        builder.try_add_source(location, instance)?;
+    }
+
+    builder.build()
+}
+
+fn make_notdef_outline(upm: u16, ascender: f64, descender: f64) -> BezPath {
```

Comment:
Optional, a comment description of what this looks like would be nice. "Draws a box around the outside plus lines between opposite corners" type of deal.